lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -6500,7 +6523,13 @@ namespace Js
             uint32 newLength = head->length + countUndefined;
             if (newLength > head->size)
             {
+                SparseArraySegmentBase *oldHead = head;
+                bool isInlineSegment = JavascriptArray::IsInlineSegment(head, this);
                 head = SparseArraySegment<Var>::From(head)->GrowByMin(recycler, newLength - head->size);
```

Comment:
>GrowByMin [](start = 60, length = 9)

Seems like this will always allocate a new segment so the check oldHead != head is not needed